xbmc/platform/darwin/tvos/TVOSTopShelf.mm

```diff
@@ -46,119 +46,140 @@
+          [categoryDict setObject:categoryTitle forKey:@"categoryTitle"];
+
+          // Create an array to store each category item
+          const int categorySize = std::min(items.Size(), MaxItems);
+          auto categoryItems = [NSMutableArray arrayWithCapacity:categorySize];
+          for (int i = 0; i < categorySize; ++i)
           {
             @autoreleasepool
             {
-              auto videoItem = videoItems.Get(i);
-              if (!videoItem->HasArt("thumb"))
-                thumbLoader.LoadItem(videoItem.get());
+              auto item = items.Get(i);
+              if (!item->HasArt("thumb"))
+                thumbLoader.LoadItem(item.get());
 
-              auto thumbnailPath = getThumbnailForItem(videoItem);
-              auto fileName = std::to_string(videoItem->GetVideoInfoTag()->m_iDbId) +
+              auto thumbnailPath = getThumbnailForItem(item);
+              auto fileName = std::to_string(item->GetVideoInfoTag()->m_iDbId) +
                               URIUtils::GetFileName(thumbnailPath);
               auto destPath = URIUtils::AddFileToFolder(raPath, fileName);
               if (!XFILE::CFile::Exists(destPath))
                 XFILE::CFile::Copy(thumbnailPath, destPath);
               else
-              {
-                // remove from array so it doesnt get deleted at the end
+                // Remove from array so it doesnt get deleted at the end
                 [filePaths removeObject:@(fileName.c_str())];
-              }
 
-              auto title = getTitleForItem(videoItem);
-              CLog::Log(LOGDEBUG, "TopShelf: - adding video to '{}' array: {}",
-                        videosKey.UTF8String, title.c_str());
-              [videosArray addObject:@{
-                @"title" : @(title.c_str()),
+              auto itemTitle = getTitleForItem(item);
+              CLog::Log(LOGDEBUG, "[TopShelf] Adding item '{}' in category '{}'",
+                        itemTitle.c_str(), categorKey.UTF8String);
+              [categoryItems addObject:@{
+                @"title" : @(itemTitle.c_str()),
                 @"thumb" : @(fileName.c_str()),
-                @"url" : @(Base64::Encode(videoItem->GetVideoInfoTag()->GetPath()).c_str())
+                @"url" : @(Base64::Encode(item->GetVideoInfoTag()->GetPath()).c_str())
               }];
             }
           }
-          [sharedDefaults setObject:videosArray forKey:videosKey];
-          sharedDictJailbreak[videosKey] = videosArray;
 
-          auto tvTitle = @(g_localizeStrings.Get(titleStringCode).c_str());
-          [sharedDefaults setObject:tvTitle forKey:videosTitleKey];
-          sharedDictJailbreak[videosTitleKey] = tvTitle;
+          // Store category items array in category dict
+          [categoryDict setObject:categoryItems forKey:@"categoryItems"];
+
+          // Store category dict in shared dict
+          if (isSandboxed)
```

Comment:
not sure why you want to make this check: the previous code without checks is fine because sending messages to `nil` is safe